buildSrc/src/main/java/org/elasticsearch/gradle/util/GradleUtils.java

```diff
@@ -170,7 +171,12 @@ public static void maybeConfigure(TaskContainer tasks, String name, Action<? sup
         });
         project.getPluginManager().withPlugin("eclipse", p -> {
             EclipseModel eclipse = project.getExtensions().getByType(EclipseModel.class);
-            eclipse.getClasspath().setSourceSets(List.of(testSourceSet));
+            List<SourceSet> eclipseSourceSets = new ArrayList<>();
+            for (SourceSet old : eclipse.getClasspath().getSourceSets()) {
```

Comment:
Can we not just do `eclipse.getClasspath().getSourceSets().add()`? Do we have to fully replace the collection?

Reply:
`getSourceSets` returns an `Iterable`!